packages/amplify-e2e-core/src/init/amplifyPush.ts

```diff
@@ -49,7 +50,15 @@ export function amplifyPush(
         }
       });
     // Test amplify push
-    const pushCommands = spawn(getCLIPath(testingWithLatestCodebase), ['push'], { cwd, stripColors: true, noOutputTimeout: pushTimeoutMS })
+    const pushOptions = ['push'];
+    if (settings?.useBetaSqlLayer) {
+      pushOptions.push('--use-beta-sql-layer');
```

Comment:
I assume this flag is hidden from all customer-facing usages (e.g., help screens, documentation, manifests, etc), so that we can use it for our E2E tests without exposing the feature to customers?

Reply:
Yes, that's correct. It is not exposed to customers.